### PR TITLE
Add better 403 error message for Job template create

### PR DESCRIPTION
**Pull request by TheRealHaoLiu**
##### SUMMARY
To create Job template u need access to projects and inventory

JIRA: https://issues.redhat.com/browse/AAP-25603

##### ISSUE TYPE

 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME

 - API

##### AWX VERSION

```

```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: awx/main/access.py [hunk 1]
```diff
@@ -1604,11 +1605,15 @@ def get_value(Class, field):
...
         # If the user has admin access to the project (as an org admin), should
         # be able to proceed without additional checks.
-        if project:
-            return self.user in project.use_role
-        else:
+        if not project:
+            return False
+
+        if self.user not in project.use_role:
+            self.messages['job_template'] = [_('You do not have use permission on Project')]
```
Comment (AlanCoding) [inline awx/main/access.py]: The field that has the error is "project" not "job_template". The user already knows they are editing a job template. So I would change that, just the _key_ here.

File: awx/main/access.py [hunk 2]
```diff
@@ -1596,6 +1596,7 @@ def get_value(Class, field):
         inventory = get_value(Inventory, 'inventory')
         if inventory:
             if self.user not in inventory.use_role:
+                self.messages['job_template'] = [_('You do not have use permission on Inventory')]
```
Comment (AlanCoding) [inline awx/main/access.py]: same here for "job_template" -> "inventory"

File: awx/api/views/__init__.py [hunk 3]
```diff
@@ -2392,6 +2392,12 @@ class JobTemplateList(ListCreateAPIView):
     serializer_class = serializers.JobTemplateSerializer
     always_allow_superuser = False
 
+    def check_permissions(self, request):
+        can_access, messages = request.user.can_access_with_errors(self.model, 'add', request.data)
+        if not can_access:
+            self.permission_denied(request, message=messages)
+        return super().check_permissions(request)
```
Comment (AlanCoding) [inline awx/api/views/__init__.py]: Oh this sucks. Thanks for digging up and figuring out this sloppy work.

**Review by AlanCoding — APPROVED** [review]

**Comment by chrismeyersfsu** [comment]
A unit/functional test would be nice here.

**Review by chrismeyersfsu — APPROVED** [review]

**Review by chrismeyersfsu — APPROVED** [review]

**Pull request merged**